service/matching/reachability.go

```diff
@@ -175,31 +184,40 @@ func (rc *reachabilityCalculator) isReachableActiveAssignmentRuleTargetOrDefault
 
 func (rc *reachabilityCalculator) existsWFAssignedToAny(
 	ctx context.Context,
-	countRequest *manager.CountWorkflowExecutionsRequest,
+	buildIdsOfInterest []string,
+	open bool,
 ) (bool, error) {
-	countResponse, err := rc.visibilityMgr.CountWorkflowExecutions(ctx, countRequest)
+	query := rc.makeBuildIdQuery(buildIdsOfInterest, open)
+
+	// try cache
+	exists, ok := rc.cache.Get(query, open)
+	if ok {
+		return exists, nil
+	}
+
+	// cache was cold, ask visibility and put result in cache
+	countResponse, err := rc.visibilityMgr.CountWorkflowExecutions(ctx, rc.makeBuildIdCountRequest(query))
```

Comment:
I guess you're planing to add hit/miss and query latency metrics for this cache. lets emit a cache size gauge as well.
Consider moving visibilityManager into reachabilityCache so all the hit/miss logic and metric business happens there.

Reply:
In the metrics PR I suggested adding a log for reachability result. that log should ideally contain tag for cache hit/miss as well.

Reply:
ok sounds good. The cache creation func takes a metrics handler as a parameter, so I want to look into what metrics it emits naturally / whether I should pre-tag that metrics handler before adding my own metrics.

Reply:
I added cache hit/miss as part of `reachabilityExitPoint` for two reasons:
   1. The point of `reachabilityExitPoint` is to give a sense of how much "work" is being done by the reachability calculation, and whether the cache was hit or not is a huge factor in that work.
   2. That means we can have the cache hit/miss metrics as part of the existing `reachabilityExitPoint` metric